### ISSUE TITLE
index out of bounds: the len is 0 but the index is 0

### ISSUE THREAD

**Issue by fergarrui** [foundry-rs/foundry#4630]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (6570a4c 2023-03-23T00:04:51.797681082Z)

### What command(s) is the bug in?

forge script

### Operating System

Linux

### Describe the bug

When using `vm.parseJsonUint` on a json where the key is not found I'm  getting:

```
The application panicked (crashed).
Message:  index out of bounds: the len is 0 but the index is 0
Location: evm/src/executor/inspector/cheatcodes/ext.rs:424

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   1: __libc_start_main<unknown>
      at /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:308

```

It happened to me having a JSON like:

```
{
  "local": {
    "prop1": 10,
  }
}

```

Used to work in previous versions:

```
vm.parseJsonUint(configJson, "local.prop1")
```

With latest I get this error:

```
Error: 
path error: 
$local.prop1
^^^^^^
```

Then I saw that `jq` syntax is required now, so when I do (it starts with a dot `.` now):

```
vm.parseJsonUint(configJson, ".local.prop1")
```

Is when I get the crash.
In my use case, I check if `prop1`  exists in some objects in the JSON, but when it is not found is when it crashes.

**Comment by fergarrui** [comment]
@mattsse Yes that's right sorry the issue happens when the property is not found.

Thanks!

**Comment by karmacoma-eth** [comment]
Wanted to confirm this, here is an alternative repro:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.17;

import {stdJson} from "forge-std/StdJson.sol";

import "forge-std/Test.sol";

contract JsonTest is Test {
    function testJson() public {
        string memory json = '{"name":"John","age":30,"city":"New York"}';

        // works
        assertEq(stdJson.readString(json, ".name"), "John");

        // crashes with
        //  The application panicked (crashed).
        //  Message:  index out of bounds: the len is 0 but the index is 0
        //  Location: evm/src/executor/inspector/cheatcodes/ext.rs:424
        stdJson.readString(json, ".darkest_secret");
    }
}
```

Running a git bisect points at f8e700ed5d605f53f2194dc8df05a0bc3a7c1e43 being the first bad commit where this started happening

**Comment by mattsse** [comment]
should be fixed via #4631
